haystack/components/classifiers/zero_shot_document_classifier.py

```diff
@@ -0,0 +1,234 @@
+    Additionally, the component can be configured to allow multiple labels to be true.
+
+    Classification is run on document's content field by default. If you want it to run on another field, set the
+    `classification_field` to one of document's meta fields.
+
+    Available models for the task of zero-shot-classification include:
+        - ``'valhalla/distilbart-mnli-12-3'``
+        - ``'cross-encoder/nli-distilroberta-base'``
+        - ``'cross-encoder/nli-deberta-v3-xsmall'``
+
+    Example usage in a pipeline that classifies documents based on predefined classification labels,
+    retrieved from a search pipeline:
+
+    ```python
+    from haystack import Document
+    from haystack.components.retrievers.in_memory import InMemoryBM25Retriever
+    from haystack.document_stores.in_memory import InMemoryDocumentStore
+    from haystack.core.pipeline import Pipeline
+    from haystack.components.classifiers import TransformersZeroShotDocumentClassifier
+
+    documents = [Document(id="0", content="Today was a nice day!"),
+                 Document(id="1", content="Yesterday was a bad day!")]
+
+    document_store = InMemoryDocumentStore()
+    retriever = InMemoryBM25Retriever(document_store=document_store)
+    document_classifier = TransformersZeroShotDocumentClassifier(
+        model="cross-encoder/nli-deberta-v3-xsmall",
+        labels=["positive", "negative"],
+    )
+
+    document_store.write_documents(documents)
+
+    pipeline = Pipeline()
+    pipeline.add_component(instance=retriever, name="retriever")
+    pipeline.add_component(instance=document_classifier, name="document_classifier")
+    pipeline.connect("retriever", "document_classifier")
+
+    queries = ["How was your day today?", "How was your day yesterday?"]
+    expected_predictions = ["positive", "negative"]
+
+    for idx, query in enumerate(queries):
+        result = pipeline.run({"retriever": {"query": query, "top_k": 1}})
+        assert result["document_classifier"]["documents"][0].to_dict()["id"] == str(idx)
+        assert (result["document_classifier"]["documents"][0].to_dict()["classification"]["label"]
+                == expected_predictions[idx])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str,
+        labels: List[str],
+        multi_label: bool = False,
+        classification_field: Optional[str] = None,
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var(["HF_API_TOKEN", "HF_TOKEN"], strict=False),
+        huggingface_pipeline_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initializes the TransformersZeroShotDocumentClassifier.
+
+        See https://huggingface.co/models for full list of available models.
+        Filter for zero-shot classification models (NLI): https://huggingface.co/models?pipeline_tag=zero-shot-classification&sort=downloads&search=nli
```

Comment:
I suppose only zero-shot models are relevant for this component.

```suggestion
        See the Hugging Face [website](https://huggingface.co/models?pipeline_tag=zero-shot-classification&sort=downloads&search=nli)
        for the full list of zero-shot classification models (NLI) models.
```